### ISSUE TITLE
2019-12-06 M2SD - remove on parameter at a time from PARAMETERLIST to identify sensitive parameters for which proper dependency management is not occurring leading to crashes

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#89]
# from 2019-12-06_M2SD.json
note that the "i" versions will be removed in an update to this MCT. They are unnecessary for SD administration.
```
mct$PARAMETERLIST <- paste(c("AUCALL","AUCALLDN","AUCDN","AUCINFO","AUCINFOC","AUCINFODN","AUCINFOi","AUCINFP","AUCINFPC","AUCINFPDN","AUCINFPi","AUCLAST","AUCLASTi","AUCLASTC","AUCLASTCi","AUCLASTDN","AUCLASTDNi","AUCT","AUCT1_T2","AUCTAUDNi","AUCTAU","AUCTAUi","AUCTDN","AUCXPCTO","AUCXPCTOi","AUCXPCTP","AUCXPCTPi","AUCXBPCTO","AUCXBPCTP","AUMCINFO","AUMCINFOi","AUMCINFP","AUMCINFPi","AUMCLAST","AUMCLASTi","AUMCTAUi","AUMCXPTO","AUMCXPTOi","AUMCXPTP","AUMCXPTPi","C0","CAVi","CEST","CLAST","CLASTi","CLO","CLOW","CLP","CLPW","CLTAUi","CLTAUWi","CMAX","CMAXi","CMAXC","CMAXCi","CMAXDN","CMAXDNi","CMIN","CMINi","CMINDN","CMINDNi","CTROUGHi","CTROUGHENDi","DIi","DOSE","DOSEi","DOSEC","DOSECi","F","FREL","FRELLAST","FRELLASTi","FTAUi","KEL","KELC0","KELNOPT","KELR","KELRSQ","KELRSQA","KELTMHI","KELTMLO","LASTTIME","LASTTIMEi","MRTIVIFO","MRTIVIFOi","MRTIVIFP","MRTIVIFPi","MRTLAST","MRTLASTi","PTF","PTFi","PTR","PTRi","PTROUGHRi","PTROUGHRENDi","RACi","RACCMAXi","RACCMINi","RSSi","TAU","TAUi","THALF","THALFF","TLAST","TLASTi","TMAX","TMAXi","TMIN","TMINi","TOLD","TOLDi","V0","VSSP","VSSPi","VSSPW","VSSPWi","VSSO","VSSOW","VSSOi","VSSOWi","VZO","VZOW","VZP","VZPW","VZTAUi","VZTAUWi","FLGACCEPTPREDOSE","FLGACCEPTTMAX","FLGACCEPTKEL","FLGACCEPTTAU"), collapse=";")
```

## i:  AUCT1_T2  removed
```
[1] "AUCTAUDNi;AUCTAU;AUCTAUi;AUCTDN;AUCXPCTO;AUCXPCTOi;AUCXPCTP;AUCXPCTPi;AUCXBPCTO;AUCXBPCTP;AUMCINFO;AUMCINFOi;AUMCINFP;AUMCINFPi;AUMCLAST;AUMCLASTi;AUMCTAUi;AUMCXPTO;AUMCXPTOi;AUMCXPTP;AUMCXPTPi;C0;CAVi;CEST;CLAST;CLASTi;CLO;CLOW;CLP;CLPW;CLTAUi;CLTAUWi;CMAX;CMAXi;CMAXC;CMAXCi;CMAXDN;CMAXDNi;CMIN;CMINi;CMINDN;CMINDNi;CTROUGHi;CTROUGHENDi;DIi;DOSE;DOSEi;DOSEC;DOSECi;F;FREL;FRELLAST;FRELLASTi;FTAUi;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;LASTTIMEi;MRTIVIFO;MRTIVIFOi;MRTIVIFP;MRTIVIFPi;MRTLAST;MRTLASTi;PTF;PTFi;PTR;PTRi;PTROUGHRi;PTROUGHRENDi;RACi;RACCMAXi;RACCMINi;RSSi;TAU;TAUi;THALF;THALFF;TLAST;TLASTi;TMAX;TMAXi;TMIN;TMINi;TOLD;TOLDi;V0;VSSP;VSSPi;VSSPW;VSSPWi;VSSO;VSSOW;VSSOi;VSSOWi;VZO;VZOW;VZP;VZPW;VZTAUi;VZTAUWi;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;FLGACCEPTTAU"
unit_conversion : Unit Class 1 (Time) time_col:  TMAX TMIN TLAST KELTMLO KELTMHI THALF LASTTIME MRTLAST MRTIVIFO MRTIVIFP  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE  parameters are scaled from  UG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 4 (Volume) volume_col:  V0 VZO VZP VSSO VSSP VSSPW  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 5 (Amount/Volume) conc_col:  C0 CMAX CMIN CLAST CMAXC CEST KELC0  parameters are scaled from  NG/ML  to  NG/ML  via scaling factor:  1 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 7 (Volume/Time) cl_col:  CLO CLP  parameters are scaled from  ML/HR  to  ML/HR  via scaling factor:  1 
Error in unit_conversion(data = data_data, map = map_data, result = computation_df,  : 
  object 'auc_unit_tmp' not found
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

##  i:  TLAST  removed
```
[1] "TLASTi;TMAX;TMAXi;TMIN;TMINi;TOLD;TOLDi;V0;VSSP;VSSPi;VSSPW;VSSPWi;VSSO;VSSOW;VSSOi;VSSOWi;VZO;VZOW;VZP;VZPW;VZTAUi;VZTAUWi;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;FLGACCEPTTAU;AUCT1_T2"
Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 10589286
In addition: There were 50 or more warnings (use warnings() to see the first 50)

```

##  i:  FLGACCEPTKEL  removed
```
[1] "FLGACCEPTTAU;AUCT1_T2;TLAST"
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 10589286
In addition: There were 50 or more warnings (use warnings() to see the first 50)

```

## i:  TLAST  removed
```
[1] "FLGACCEPTKEL"
Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 10589286
In addition: There were 33 warnings (use warnings() to see them)
```

## CMAX/TMAX/C0/AUCLAST
```
> mct$PARAMETERLIST <- "CMAX;TMAX;C0;AUCLAST"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 10589286
In addition: Warning messages:
1: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 10589286 
```

**Comment by jhhughes256** [comment]
### Update

Testing results for above examples as of commit fce7a3c. All testing was done with tc013 M2SD. Tick represents success (i.e. error can no longer be replicated).

- [ ] AUCT1_T2 removed "AUCTAUDNi;AUCTAU;AUCTAUi;AUCTDN;AUCXPCTO;AUCXPCTOi;AUCXPCTP;AUCXPCTPi;AUCXBPCTO;AUCXBPCTP;AUMCINFO;AUMCINFOi;AUMCINFP;AUMCINFPi;AUMCLAST;AUMCLASTi;AUMCTAUi;AUMCXPTO;AUMCXPTOi;AUMCXPTP;AUMCXPTPi;C0;CAVi;CEST;CLAST;CLASTi;CLO;CLOW;CLP;CLPW;CLTAUi;CLTAUWi;CMAX;CMAXi;CMAXC;CMAXCi;CMAXDN;CMAXDNi;CMIN;CMINi;CMINDN;CMINDNi;CTROUGHi;CTROUGHENDi;DIi;DOSE;DOSEi;DOSEC;DOSECi;F;FREL;FRELLAST;FRELLASTi;FTAUi;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;LASTTIMEi;MRTIVIFO;MRTIVIFOi;MRTIVIFP;MRTIVIFPi;MRTLAST;MRTLASTi;PTF;PTFi;PTR;PTRi;PTROUGHRi;PTROUGHRENDi;RACi;RACCMAXi;RACCMINi;RSSi;TAU;TAUi;THALF;THALFF;TLAST;TLASTi;TMAX;TMAXi;TMIN;TMINi;TOLD;TOLDi;V0;VSSP;VSSPi;VSSPW;VSSPWi;VSSO;VSSOW;VSSOi;VSSOWi;VZO;VZOW;VZP;VZPW;VZTAUi;VZTAUWi;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;FLGACCEPTTAU"
- [ ] TLAST removed "TLASTi;TMAX;TMAXi;TMIN;TMINi;TOLD;TOLDi;V0;VSSP;VSSPi;VSSPW;VSSPWi;VSSO;VSSOW;VSSOi;VSSOWi;VZO;VZOW;VZP;VZPW;VZTAUi;VZTAUWi;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;FLGACCEPTTAU;AUCT1_T2"
- [ ] FLGACCEPTKEL removed "FLGACCEPTTAU;AUCT1_T2;TLAST"
- [ ] TLAST removed "FLGACCEPTKEL"
- [ ] "CMAX;TMAX;C0;AUCLAST"

**Comment by rudraya** [comment]
This issue was addressed in a377a56, please let me know if it still occurs. I tested again using the parameters list provided in the above comment for when:
- AUCT1_T2 removed

**Comment by jhhughes256** [comment]
### Update

As of a377a56 all test cases listed here run without computation error.
